app/controllers/course/assessment/question/multiple_responses_controller.rb

```diff
@@ -25,7 +25,11 @@ def edit
   end
 
   def update
-    if @multiple_response_question.update_attributes(multiple_response_question_params)
+    @question_assessment = load_question_assessment_for(@multiple_response_question)
```

Comment:
If this is going to be called for each question type, why not just move it into a callback?

Reply:
Moved into a `before_action` filter, similar to what was done for the Programming question controller. Is that what you meant?

Reply:
Can we move that `before_action` callback into the parent controller, since it is called for all question types?

Reply:
I tried that but don't currently have a way to get the resource for each question type as they're all named differently.

Reply:
You can define a new method `specific_question`, enforce that all controllers implement it, then `load_question_assessment_for(specific_question)`?

Reply:
Just tried and it doesn't work because `load_and_authorize_resource` hasn't been called yet. `load_and_authorize_resource` also works with `before_action` callbacks and `append_before_action` [doesn't actually do anything](https://stackoverflow.com/questions/28147359/whats-difference-between-append-before-action-and-before-action-in-rails).

So when `load_question_assessment` in the parent is called and it tries to get `specific_question`, it hasn't been loaded yet and so `question_assessment` can't be retrieved.

Reply:
Do it after [this line](https://github.com/Coursemology/coursemology2/pull/2999/files#diff-3cc3e66ef644637c46dd7999144df581R20)?

Reply:
As discussed offline, this only applies to new questions. However, using a QuestionAssessment controller to handle this instead of the Question controller (as suggested in the main review comments) might give the same result.